apps/create/widget/Provider.jsx

```diff
@@ -1,211 +1,311 @@
 /*__@import:everything/utils/UUID__*/
+/*__@import:QoL/storage__*/
+
+const DEBUG = true;
 
 const Children = props.Children;
 
 const KEYS = {
   selectedDoc: (pid) => `selectedDoc/${pid}`,
-  doc: (did) => `doc/${did}`,
-  docs: (pid) => `docs/${pid}`, // project' docs structure without content
+  doc: (path) => `doc/${path}`, // having path here will let us always know the structure of the doc
+  docs: (pid) => `docs/${pid}`, // this should be the array of project docs
 };
+const DOC_SEPARATOR = ".";
 
-const store = (k, v) => Storage.privateSet(k, v);
-const retrieve = (k) => Storage.privateGet(k);
-
-const setPath = (docs, path, value) => {
-  if (path.length === 1) {
-    if (value !== null) {
-      const updatedDoc = {
-        ...docs[path[0]],
-        ...value,
-      };
-      return {
-        ...docs,
-        [path[0]]: updatedDoc,
-      };
-    } else {
-      // We're deleting a document
-      const remainingDocs = JSON.parse(JSON.stringify(docs));
-      delete remainingDocs[path[0]];
-      return remainingDocs;
-    }
-  } else {
-    docs[path[0]].children = setPath(
-      docs[path[0]].children,
-      path.slice(1),
-      value
-    );
-    return docs;
-  }
-};
-const getPath = (docs, path) => {
-  return path.reduce((currentDoc, key, index) => {
-    // If it's the last key in the path, return the document itself
-    if (index === path.length - 1) {
-      return currentDoc[key];
-    }
-    // Otherwise, navigate to the children of the current document
-    return currentDoc[key].children;
-  }, docs);
-};
-const didToPath = (docs, id) => {
-  const path = [];
-  const traverse = (docs, id) => {
-    for (const key in docs) {
-      if (key === id) {
-        path.push(key);
-        return true;
-      }
-      if (docs[key].children) {
-        path.push(key);
-        if (traverse(docs[key].children, id)) {
-          return true;
-        }
-        path.pop();
+/**
+ * 
+ * I'm trying to learn what is the best way to structure our data, 
```

Comment:
Yeah, I was coming to a similar conclusion last night that it should be flattened. I like this so far, let's try it